src/vs/workbench/contrib/extensions/electron-browser/extensionsViewlet.ts

```diff
@@ -439,23 +442,36 @@ export class ExtensionsViewlet extends ViewContainerViewlet implements IExtensio
 		this.nonEmptyWorkspaceContextKey.set(this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY);
 
 		if (value) {
-			return this.progress(Promise.all(this.panels.map(view =>
-				(<ExtensionsListView>view).show(this.normalizedQuery())
-					.then(model => this.alertSearchResult(model.length, view.id))
-			)));
+			return this.searchOverViews(this.panels);
 		}
 		return Promise.resolve(null);
 	}
 
 	protected onDidAddViews(added: IAddedViewDescriptorRef[]): ViewletPanel[] {
 		const addedViews = super.onDidAddViews(added);
-		this.progress(Promise.all(addedViews.map(addedView =>
-			(<ExtensionsListView>addedView).show(this.normalizedQuery())
-				.then(model => this.alertSearchResult(model.length, addedView.id))
-		)));
+
+		this.searchOverViews(addedViews);
```

Comment:
I do not think you should see this error when you override `onDidRemoveViews` from super class

https://github.com/Microsoft/vscode/blob/3d662dd0e1eb1e89e3dd10c3235a8f9a87f58e53/src/vs/workbench/browser/parts/views/viewsViewlet.ts#L226

Did you try changing it to protected and overriding it in ExtensionsViewlet?